Adds 2-step migration tuning serializers (#2285)
lagom/lagom: docs/manual/java/releases/Migration16.md

Lagom is now using the Jackson serializer from Akka, which is an improved version of the serializer in Lagom 1.5. You can find more information about the Akka Jackson serializer in the [Akka documentation](https://doc.akka.io/docs/akka/2.6/serialization-jackson.html). It is compatible with Lagom 1.5 in both directions.

> The serializer provided by Akka creates new [Akka serialization bindings](https://doc.akka.io/docs/akka/current/serialization.html) so if you want to do a rollingupgrade from Lagom 1.5.x to Lagom 1.6.x you need to do a [[multi-step upgrade|Migration16]]. 

Reviewer: ```suggestion
The serializer provided by Akka creates new [Akka serialization bindings](https://doc.akka.io/docs/akka/current/serialization.html) so if you want to do a rolling upgrade from Lagom 1.5.x to Lagom 1.6.x you need to do a [[multi-step upgrade|Migration16]]. 
```

* The change in [[Akka Remote|Migration16#Remoting-Artery]] default implementation.
* The change in default [[Shard Coordination|Migration16#Shard-Coordination]] strategy.
* The change in [[Cassandra plugin version|Migration16#Akka-Persistence-Cassandra-Update]]. Only impact Lagom applications using Cassandra.

### Multi-step upgrade: serialization

> Changes in default serialisers require rollingupgrades to happen in two steps.

Reviewer: ```suggestion
Changes in default serializers require rolling upgrades to happen in two steps.
```

> #### Step 1: new binary, old serialisers

Reviewer: ```suggestion
#### Step 1: new binary, old serializers
```

> First, you will have to deploy a version of your binary that's already a Lagom 1.6.x application but you must set it up to use the serialisers that are compatible with Lagom 1.5.x so you old nodes can continu to operate. On you `application.conf` use:

Reviewer: ```suggestion
First, you will have to deploy a version of your binary that's already a Lagom 1.6.x application but you must set it up to use the serializers that are compatible with Lagom 1.5.x so you old nodes can continue to operate. On you `application.conf` use:
```

```
## Lagom 1.6 no longer enables Java serialization by default but some 
## code in Lagom 1.5 (or your own code) still used it so we enable it back.

> akka.actor.allow-java-serialization = on

Reviewer 2: Is that true? I'm surprised that we had internal code using java serialization.

Author: I didn't track it down. It's most likely user code on the test app I used and the logs didn't point to any class in particular. I checked the code and everything looked fine.

I just checked again and noticed [`Get`](https://github.com/renatocaval/lagom-1.6-playground/blob/1351fc2561d7bf59074e9f17c79c3ffc9bb3c1af/shopping-cart-java/node1/shopping-cart/src/main/java/com/example/shoppingcart/impl/ShoppingCartCommand.java#L46) so maybe that was it.

In any case, Lagom or user code, with this trick you are safe (as long as the offending class is not an event serializing data in Java into the journal). Alternatively, users would have to do the serializer dance (from Java to JSON in a Lagom 1-5-based version of their code) and once all the deployed nodes use JSON only, _then_ start migrating to Lagom 1.6.

## Configures Akka to only produce messages using the `old-lagom-json` serializer
akka.actor {
  serialization-bindings {
    "com.lightbend.lagom.serialization.Jsonable" = old-lagom-json
    "com.lightbend.lagom.serialization.CompressedJsonable" = old-lagom-json
  }
}
```

The setup above makes all traffic and durable data produced by Lagom 1.6 code to be readable by a Lagom 1.5 binary.

#### Step 2: new binary, default serializers

> Once all your productio nodes are running a binary of you service based on Lagom 1.6 you can remove the settings for the old serializers from `application.conf` and use the defaults.

Reviewer: ```suggestion
Once all your production nodes are running a binary of you service based on Lagom 1.6 you can remove the settings for the old serializers from `application.conf` and use the defaults.
```

> Once all your productio nodes are running a binary of you service based on Lagom 1.6 you can remove the settings for the old serializers from `application.conf` and use the defaults.

Reviewer: ```suggestion
Once all your production nodes are running a binary of your service based on Lagom 1.6 you can remove the settings for the old serializers from `application.conf` and use the defaults.
```
